deeplinkdispatch-base/src/test/java/com/airbnb/deeplinkdispatch/DeepLinkMatchTests.kt

```diff
@@ -46,15 +46,21 @@ class DeepLinkMatchTests {
         }
     }
 
-    @Test(expected = IllegalStateException::class)
+    @Test
     fun testMatchArraySerializationDeserializationNonExistantClass() {
-        val matchByteArray = matchByteArray(UriMatch(ONE_PARAM_SCHEMA, "soneNonexistantClass", null))
-        MatchIndex(matchByteArray.toByteArray()).getMatchResultFromIndex(
+        val matchByteArray = matchByteArray(UriMatch(ONE_PARAM_SCHEMA, "someNonexistantClass", null))
+        val entryFromArray = MatchIndex(matchByteArray.toByteArray()).getMatchResultFromIndex(
             matchByteArray.size,
             0,
             DeepLinkUri.parse("schema://none"),
             emptyMap()
         )
+        assertNotNull(entryFromArray)
+        entryFromArray?.let {
```

Comment:
@pmecho They will, I do null check `entryFromArray` just above. So if it would actually be null we would have already failed the test. But I guess that would be even safer to use !!